### PR TITLE
sync timingDiagram with Cromwell 34 JS [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
Context/notes to reviewer: For a bunch of infrastructure/network reasons, FireCloud implements its own JavaScript for rendering workflow timing diagrams of Cromwell analyses. FireCloud's JS is, for the most part, a copy/paste of the JS from Cromwell's timing endpoint (see https://cromwell1.dsde-dev.broadinstitute.org/swagger/index.html?url=/swagger/cromwell.yaml#/Workflows/timing).

However, it's been a while since FireCloud synced up that JavaScript. Cromwell has implemented improvements since our last sync. This PR attempts to get FireCloud up to speed with current Cromwell (34).

To review:
1) grab the JavaScript from https://cromwell1.dsde-dev.broadinstitute.org/swagger/index.html?url=/swagger/cromwell.yaml#/Workflows/timing
2) Compare the changes in this PR to Cromwell's JS

I've made one functional change, which is to strip the workflow-name prefix from each row of the timing diagram. This matches what Cromwell does currently, but changes behavior compared to current FireCloud. Visually, the change is:

_current FireCloud_
timing-before

_after this PR_
timing-after

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews

  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
multiple failures because - I'm pretty sure - gpalloc was redeployed during my test run. Let's just do it again: jenkins retest.

File: src/js/timingDiagram.js [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
+/*
+    following code, up until the END marker, is copied directly from Cromwell WITH THREE
+    EXCEPTIONS. Each of the three exceptions is preceded by a comment. These comments
+    all start with "DA:" and end with "for FireCloud."
+*/
+var parentWorkflowNames = [];
+var expandedParentWorkflows = [];
+var chartView;
+
+function addDataTableRow(dataTable, callName, callPartName, startDate, endDate, ancestry) {
```
Comment (ansingh7115) [inline src/js/timingDiagram.js:17]: Maybe a quick comment somewhere about what ancestry is?

File: src/js/timingDiagram.js [hunk 2]
```diff
@@ -95,43 +102,159 @@ function timingDiagram(element, response, workflowName, height) {
...
             avoidOverlappingGridLines: false,
             showBarLabels: false,
-            rowLabelStyle: { fontName: 'Roboto', fontSize: 12, color: '#333' },   //font-size 10   #603913
+            rowLabelStyle: { fontName: 'Roboto', fontSize: 12, color: '#333' },
             // Although bar labels are unshown, they still affect the height of each row. So make it small.
-            barLabelStyle: { fontName: 'Roboto', fontSize: 8, color: '#333' },    //#603913
-        }
+            barLabelStyle: { fontName: 'Roboto', fontSize: 8, color: '#333' },
+        },
+        tooltip: {isHtml: true}
```
Comment (ansingh7115) [inline src/js/timingDiagram.js:232]: Question: does this change the way our timing diagram tooltips are implemented?

**Comment by ansingh7115** [comment]
Is this built on a fiab somewhere?

File: src/js/timingDiagram.js (hunk 2 above)
Comment (davidangb) [reply src/js/timingDiagram.js:232]: not that I can tell. `isHtml: true` is the default, according to https://developers.google.com/chart/interactive/docs/gallery/timeline#configuration-options, so this should be a noop.

File: src/js/timingDiagram.js (hunk 1 above)
Comment (davidangb) [reply src/js/timingDiagram.js:17]: added a comment at line 9.

**Comment by davidangb** [comment]
@ansingh7115 I added a comment and replied about the tooltips. I do not have this running in a fiab anywhere; I just ran the UI locally to view it.

**Comment by davidangb** [comment]
jenkins retest

**Review by ansingh7115 — APPROVED** [review]
Looks good to me! Disclaimer: I don't have a lot of JS experience, but it's similar to what Cromwell already implements and I agree with the places that differ.

**Comment by davidangb** [comment]
automated tests had one failure, which looks transient. I'll retry just that one.
```
[info] *** 1 TEST FAILED ***
[error] Failed: Total 123, Failed 1, Errors 0, Passed 122, Ignored 7
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.analysis.MethodLaunchSpec
[error] (Test / testOnly) sbt.TestsFailedException: Tests unsuccessful

[info] - launch a method from the method repo *** FAILED *** (3 minutes, 2 seconds)
[info]   org.openqa.selenium.remote.UnreachableBrowserException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
```

EDIT: it passed, in https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/11622/.

**Pull request merged**